### PR TITLE
accept array of case_type_ids when finding available case statuses

### PR DESCRIPTION
**Pull request by ufundo**
Overview
----------------------------------------
Chain selects from case_type_id => case status are broken because the Case BAO ~~breaks~~ ignores when passed an array of case_type_ids.

Example afform for a search of Cases 

```
<div af-fieldset="">
  <af-field name="case_type_id" defn="{input_attrs: {multiple: false}}" />
  <af-field name="status_id" defn="{input_type: 'ChainSelect', input_attrs: {multiple: true, control_field: 'case_type_id'}}" />

  <crm-search-display-table search-name="Case_Search_by_Standalone_Admin" display-name=""></crm-search-display-table>
</div>
```

Before
----------------------------------------
The case statuses aren't restricted. 

After
----------------------------------------
Works as expected (with https://github.com/civicrm/civicrm-core/pull/29662 )

Comments
----------------------------------------
- I originally hit this on 5.69 where it exploded rather than just ignored (before the is_scalar check was introduced)... so less of an issue now.
- @colemanw  expressed concern about the cost of unserializing all the Case Type definitions from XML... see discussion here: https://github.com/civicrm/civicrm-core/pull/28957
- wouldn't it be dreamy if the definition wasn't stored in XML? or could implement some cacheing at API4 layer? obviously a bigger undertaking...

### CODE REVIEW

File: CRM/Case/BAO/Case.php [hunk 1]
```diff
@@ -2995,11 +2995,28 @@ public static function buildOptions($fieldName, $context = NULL, $props = []) {
...
       // Filter status id by case type id
       case 'status_id':
-        if (!empty($props['case_type_id']) && is_scalar($props['case_type_id'])) {
-          $idField = is_numeric($props['case_type_id']) ? 'id' : 'name';
-          $caseType = civicrm_api3('CaseType', 'getsingle', [$idField => $props['case_type_id'], 'return' => 'definition']);
-          if (!empty($caseType['definition']['statuses'])) {
-            $params['condition'] = 'v.name IN ("' . implode('","', $caseType['definition']['statuses']) . '")';
+        if (!empty($props['case_type_id'])) {
+          // cast single values to a single value array
+          $caseTypeIdValues = is_array($props['case_type_id']) ? $props['case_type_id'] : [$props['case_type_id']];
```
Comment (colemanw) [inline CRM/Case/BAO/Case.php:3000]: Could just be 
```suggestion
          $caseTypeIdValues = (array) $props['case_type_id'];
```

File: CRM/Case/BAO/Case.php [hunk 2]
```diff
@@ -2995,11 +2995,28 @@ public static function buildOptions($fieldName, $context = NULL, $props = []) {
...
+
+          $idField = is_numeric($caseTypeIdValues[0]) ? 'id' : 'name';
+          $caseTypeDefs = (array) \Civi\Api4\CaseType::get(TRUE)
```
Comment (colemanw) [inline CRM/Case/BAO/Case.php:3003]: This flips the permission switch (the old code used api3 which defaults to FALSE.
I'm not exactly sure which is more correct but would recommend keeping it consistent so changing this to
```suggestion
          $caseTypeDefs = (array) \Civi\Api4\CaseType::get(FALSE)
```

**Comment by ufundo** [comment]
accepted those suggestions @colemanw - thanks!

how do you feel about the performance issue?

**Comment by ufundo** [comment]
Rebased onto master because it includes merged https://github.com/civicrm/civicrm-core/pull/29662

**Comment by colemanw** [comment]
@ufundo looks good. I think the performance thing might need to be revisited but not a blocker to this PR.

**Pull request merged**